fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisJob.java

```diff
@@ -187,7 +187,7 @@ public void cancel() {
     }
 
     public void deregisterJob() {
-        analysisManager.removeJob(jobInfo.jobId);
+        analysisManager.analysisJobIdToTaskMap.remove(jobInfo.jobId);
```

Comment:
we'd better add back removeJob method in AnalysisManager, and change the code inner AnalysisManager#removeJob